db/dcrpg/queries.go

```diff
@@ -724,30 +724,47 @@ func RetrieveAddressSpentUnspent(db *sql.DB, address string) (numSpent, numUnspe
 		return
 	}
 
-	var nu, tu sql.NullInt64
-	err = dbtx.QueryRow(internal.SelectAddressUnspentCountANDValue, address).
-		Scan(&nu, &tu)
+	// Query for spent and unspent totals
+	var rows *sql.Rows
+	rows, err = db.Query(internal.SelectAddressSpentUnspentCountAndValue, address)
 	if err != nil && err != sql.ErrNoRows {
 		if errRoll := dbtx.Rollback(); errRoll != nil {
 			log.Errorf("Rollback failed: %v", errRoll)
 		}
-		err = fmt.Errorf("unable to QueryRow for unspent amount: %v", err)
+		err = fmt.Errorf("unable to Query for spent and unspent amounts: %v", err)
+		return
+	}
+	if err == sql.ErrNoRows {
 		return
 	}
-	numUnspent, totalUnspent = nu.Int64, tu.Int64
 
-	var ns, ts sql.NullInt64
-	err = dbtx.QueryRow(internal.SelectAddressSpentCountANDValue, address).
-		Scan(&ns, &ts)
-	if err != nil && err != sql.ErrNoRows {
-		if errRoll := dbtx.Rollback(); errRoll != nil {
-			log.Errorf("Rollback failed: %v", errRoll)
+	for rows.Next() {
+		var count, totalValue int64
+		var noMatchingTx, isFunding bool
+		err = rows.Scan(&count, &totalValue, &isFunding, &noMatchingTx)
+		if err != nil {
+			break
```

Comment:
This will exit the for loop without giving any notification to as why the loop exited prematurely.